pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -783,26 +770,52 @@ def append_tk_command_to_qiskit(
         if optype == OpType.CustomGate:
             instruc = subqc.to_gate()
             instruc.name = op.get_name()
+            _apply_qiskit_instruction(qcirc, instruc, qargs, condition)  # type: ignore
         else:
-            instruc = subqc.to_instruction()
-        return qcirc.append(instruc, qargs, cargs)
-    if optype in [OpType.Unitary1qBox, OpType.Unitary2qBox, OpType.Unitary3qBox]:
+            if _has_control_flow(subqc):
+                # Detect control flow in CircBoxes and raise an error.
+                raise NotImplementedError(
+                    "Conversion of CircBox(es) containing conditional"
+                    + " gates not currently supported by tk_to_qiskit"
+                )
+            else:
+                instruc = subqc.to_instruction()
+                _apply_qiskit_instruction(qcirc, instruc, qargs, condition)  # type: ignore
```

Comment:
Would be nice to remove the `# type: ignore` here and on line 773

Not sure why mypy doesn't like the type of the `condition` arg to `_apply_qiskit_instruction`... It should be the same type as the input to `append_tk_command_to_qiskit`.

```
Argument 4 to "_apply_qiskit_instruction" has incompatible type "tuple[Any | Any, int] | None"; expected "Iterable[Any]"  [arg-type]
```

Reply:
Argument 4 is `cargs`. This seems to be missing from the call here (and above). Better to use named arguments to avoid this kind of bug.